src/scheduler.py

```diff
@@ -197,10 +197,10 @@ def UpdateClusterState(self, request, context):
                 updated_state['mps'].num_available -= 1
         # Log completed runs to retrieve stats post processing from kafka
         elif request.reason == 'end mix':
-            echo_cmd = f"echo {request.run_id} >> /tmp/tiebreaker-completed-mixes.txt"
+            echo_cmd = f'[ -p "/tmp/{request.run_id}.client-wait" ] && echo "1" > /tmp/{request.run_id}.client-wait'
```

Comment:
For each client, we will write into a fifo file that the client creates. If there is no file, then scheduler does nothing. This both the logics are simple.

Reply:
Great, thanks for this change.